core/src/main/scala/io/finch/internal/package.scala

```diff
@@ -122,4 +124,12 @@ package object internal {
     }
   }
 
+  def futureToEffect[F[_] : Effect, A](future: => Future[A]): F[A] = {
```

Comment:
I like idea. We needed it in a couple of other places so better to reuse it.

What do you think about a type class `LiftFuture` (by analogy with https://typelevel.org/cats-effect/typeclasses/liftio.html) that would let us lift both Scala futures and Twitter Futures into `F[_]: Effect`. I'd imagine we can simplify some code in Mapper with it.

Reply:
Yeah, it's possible

Reply:
Should we rebase this PR as we now have `ToEffect`?

Reply:
I just realized that `ToEffect` we have now is not lazy on the argument, so `Future` got executed immediately instead of being suspended in `Effect.async`.

Reply:
Here is what I mean:

```
import cats.effect.IO
import com.twitter.util.{Future, FuturePool}
import fs2._
import io.finch.internal._

def future: Future[Int] = FuturePool.unboundedPool {
  val i = 5
  println(i)
  i
}

Stream.repeatEval(futureToEffect[IO, Future, Int](future)) //it prints 5, but we didn't run stream yeat
```

Reply:
Yes, we can. But it would mean that we penalize the lawful effect uses with the additional allocation of `Function0` in `Mapper`. Actually, we're doing it already.

https://github.com/finagle/finch/blob/master/core/src/main/scala/io/finch/internal/Mapper.scala#L83-L84

https://github.com/finagle/finch/blob/master/core/src/main/scala/io/finch/internal/Mapper.scala#L83-L84

Shall we specialize it to make Futures called-by-name and `E : Effect` to be called by value?
Sorry, that was a too late observation from my side 😞

Reply:
I'm starting to think the signature of `ToEffect` may not be out problem. These got evaluated only when the request comes in anyway, and more importantly, only when `Endpoint.map*` is being run. When dealing with `Endpoint[IO, ?]`, for example, this only happens when we run the endpoint at the end, which pretty much neglects the difference between eager/lazy evaluation of that type-class.

But I agree, we should strive for consistency. I'm not super sure though, it's possible to get rid of call-by-name args in mapper. This [very simple test](https://github.com/finagle/finch/blob/master/core/src/test/scala/io/finch/EndpointSpec.scala#L263-L270) would likely to fail.

Reply:
@vkostyukov 
Probably, I've explained it not clear enough.

So my point is:
- Lawful effects `F[_]` _should not_ be called by name. We run them inside of Service, but it's enough to instantiate this effect once.
- Futures should be _should_ be called by name, because evaluation starts once Future is instantiated.
- Responses & Outputs _should_ be called by name, for the very same reason as for Future.

Reply:
Yeah, I get it. That's my point, even with `F[_]: Effect` and call-by-value, the linked test would fail - give it a try.

Reply:
Should be resolved by now